File: OrangeJuice.Server.Core/Configuration/AwsOptionsFactory.cs
using System;

namespace OrangeJuice.Server.Configuration
{
	public class AwsOptionsFactory : Factory.IFactory<AwsOptions>
	{
		private readonly IConfigurationProvider _configurationProvider;

		public AwsOptionsFactory(IConfigurationProvider configurationProvider)
		{
			_configurationProvider = configurationProvider;
		}

		public AwsOptions Create()
		{
			return new AwsOptions
			{
				AccessKey = _configurationProvider.GetValue("aws:AccessKey"),
				AssociateTag = _configurationProvider.GetValue("aws:AssociateTag"),
				SecretKey = _configurationProvider.GetValue("aws:SecretKey"),
				RequestLimit = TimeSpan.Parse(_configurationProvider.GetValue("aws:RequestLimit"))
			};
		}
	}
}

File: OrangeJuice.Server.Core.FSharp/Configuration/AwsOptionsFactory.fs
namespace OrangeJuice.Server.FSharp.Configuration

open System
open OrangeJuice.Server.Configuration

type AwsOptionsFactory(configurationProvider : IConfigurationProvider) =
    interface Factory.IFactory<AwsOptions> with
        member this.Create() : AwsOptions =
            new AwsOptions(AccessKey = configurationProvider.GetValue("aws:AccessKey"),
                           AssociateTag = configurationProvider.GetValue("aws:AssociateTag"),
                           SecretKey = configurationProvider.GetValue("aws:SecretKey"),
                           RequestLimit = TimeSpan.Parse(configurationProvider.GetValue("aws:RequestLimit")))
